core/src/main/java/pl/agh/edu/model/HotelPopularityFunction.java

```diff
@@ -14,9 +17,23 @@
 
 public class HotelPopularityFunction {
 
-	public static HashMap<MonthDay, Double> getSeasonalMultipliers(HotelType hotelType) {
+	public static Map<MonthDay, Double> getSeasonalMultipliers(HotelType hotelType) {
+		PolynomialSplineFunction splineFunction = getSplineFunction(hotelType);
+		return getDailyMultiplierFromSpline(splineFunction);
+	}
+
+	private static Map<MonthDay, Double> getDailyMultiplierFromSpline(PolynomialSplineFunction splineFunction) {
+		return Stream.of(Month.values())
+				.flatMap(month -> IntStream.range(1, month.maxLength() + 1).mapToObj(day -> MonthDay.of(month, day)))
+				.collect(Collectors.toMap(
+						Function.identity(),
+						monthDay -> splineFunction.value(monthDay.getMonth().getValue() + (monthDay.getDayOfMonth() - 1.) / monthDay.getMonth().maxLength()),
+						(a, b) -> b,
+						HashMap::new));
+	}
+
+	private static PolynomialSplineFunction getSplineFunction(HotelType hotelType) {
```

Comment:
to bym tak samo jeszcze podzielił na mniejsze częsci coś w stylu (nie wiem czy to będzie działać )
```java 
private static PolynomialSplineFunction getSplineFunction(HotelType hotelType) {
    Map<Integer, Double> monthClientsMultiplier = JSONHotelScenariosDataLoader.vacationPopularity.get(hotelType);
    double[] x = generateXValues();
    double[] y = generateYValues(monthClientsMultiplier);
    
    return createSplineFunction(x, y);
}

private static double[] generateXValues() {
    return IntStream.rangeClosed(0, 13)
            .asDoubleStream()
            .toArray();
}

private static double[] generateYValues(Map<Integer, Double> monthClientsMultiplier) {
    return IntStream.range(0, 14)
            .mapToDouble(i -> calculateYValue(monthClientsMultiplier, i))
            .toArray();
}

private static double calculateYValue(Map<Integer, Double> monthClientsMultiplier, int i) {
    int month = (i % 12) + 1;
    return monthClientsMultiplier.getOrDefault(month, 0.0);
}

private static PolynomialSplineFunction createSplineFunction(double[] x, double[] y) {
    SplineInterpolator interpolator = new SplineInterpolator();
    return interpolator.interpolate(x, y);
}
```